cfi/lib/src/cfi.rs

```diff
@@ -94,16 +94,75 @@ impl From<CfiPanicInfo> for CaliptraError {
 /// # Returns
 ///
 /// `T` - Same value
-pub fn cfi_launder<T>(val: T) -> T {
+pub fn cfi_launder<T>(val: T) -> T
+where
+    Launder<T>: LaunderTrait<T>,
+{
     if cfg!(feature = "cfi") {
-        // Note: The black box seems to be disabling more optimization
-        // than necessary and results in larger binary size
-        core::hint::black_box(val)
+        Launder { _val: PhantomData }.launder(val)
     } else {
         val
     }
 }
 
+pub trait LaunderTrait<T> {
```

Comment:
These Rust type acrobatics are so that we can have special, smaller implementations for register-sized but default to `core::hint::black_box` for everything else.

I'm open to suggestions on how to make this simpler or avoid a `derive`.